solr/core/src/java/org/apache/solr/search/SolrReturnFields.java

```diff
@@ -50,6 +51,7 @@
 public class SolrReturnFields extends ReturnFields {
   // Special Field Keys
   public static final String SCORE = "score";
+  public static final String MATCH_SCORE = "matchScore";
```

Comment:
I'm not loving another special case, vs a special function like say `matchScore()`.  If we were starting Solr from scratch, even `score` would be `score()` IMO.

Reply:
I agree, but at the time score() didn't exist (and still doesn't yet I guess). Happy to eventually move over to that, and deprecate this when it can be implemented well. (Discussion on https://github.com/apache/solr/pull/3340 probably makes more sense)

Reply:
I think this part of the change bothers me the most so I'm looking forward to PR3340 to hopefully make this obsolete.

Reply:
So we _could_ set this to be "matchScore()" as a special case, and eventually change it to be a function when that can be supported better.

Reply:
That would at least unblock us for a release, so that we don't have to go and deprecate something later.

Reply:
+1 with a TODO JIRA link